modules/url-titles.js

```diff
@@ -0,0 +1,47 @@
+const cheerio = require('cheerio')
+const request = require('request')
+var getTitles = function (bot, to, nick, urls, titles) {
+  const maxSize = 1024
+  var size = 0
+  var response = ''
+  var req = request(urls[0])
+  // Check the size of the downloaded data regularly
+  req.on('data', chunk => {
```

Comment:
Can we explicitly set the `highWaterMark` in request's [ReadableStream](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/ReadableStream)? (I'm not sure.)
At the moment, a chunk received from the `data` event can be considerably larger than 1024 bytes.